packages/flutter_tools/lib/src/compile.dart

```diff
@@ -662,7 +662,7 @@ class ResidentCompiler {
   }
 
   Future<dynamic> shutdown() {
-    _server.kill();
-    return _server.exitCode;
+    _server?.kill();
+    return _server?.exitCode;
```

Comment:
Is it meaningful to return `null` if `_server` is `null`? Maybe return `0`? If it an unexpected situation, perhaps throw or print a warning?

Reply:
Done